wagtail/api/v2/filters.py

```diff
@@ -34,7 +34,7 @@ def filter_queryset(self, request, queryset, view):
                 try:
                     if isinstance(field, (models.BooleanField, models.NullBooleanField)):
                         value = parse_boolean(value)
-                    elif isinstance(field, (models.IntegerField, models.AutoField)):
+                    elif isinstance(field, (models.IntegerField, models.ForeignKey, models.AutoField)):
```

Comment:
``ForeignKey``'s could sometimes be text or a UUID internally.

Could you try using `field.get_prep_value(value)` here? this should raise a `ValueError` if the value is invalid

Reply:
I thought about it but I didn't know how to handle it.
I made the changes requested.
Is that it?

Reply:
Tests aren't passing for mysql 3.6 appearly. 
How can it be fixed?

Reply:
I'm not sure what's going on there, maybe we could try performing `get_prep_value` on the remote field?

So try: `value = field.remote_field.get_prep_value(value)`

Reply:
Seems that it fixes the problem. However the error message for mysql 3.65 is different which makes the tests fail. How can that be fixed? 
If we only make assertions for status code, tests should pass but it is the right solution?

Reply:
Maybe we could test the prefix? ``self.assertEqual(content['message'][:62], 'field filter error. 'abc' is not a valid value for feed_image')``

Reply:
Tests pass now!

Reply:
Nice one, thanks!